### PR TITLE
add workflows

### PR DESCRIPTION
**Pull request by Monirzadeh**
try to review workflows in all `go-shiori` projects and unify them for easier maintenance in long term. 
- [x] add golangci-lint workflow
- [x] add unittest workflow
- [x] add dependabot

### CODE REVIEW

**Comment by Monirzadeh** [comment]
I am not sure if I am running UnitTest in Workflow correctly.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    - name: Set up gotestfmt
+      uses: GoTestTools/gotestfmt-action@v2
+
+    - name: Run tests
+      run: |
+        set -euo pipefail
+        go test -json -v ./... 2>&1 | tee /tmp/gotest.log | gotestfmt
+     
+        # Upload the original go test log as an artifact for later review.
```
Comment (fmartingr) [inline .github/workflows/ci.yml]: Do we really need this? We can see the output in the action logs and download it, and I don't know if we would ever go back to past tests and download the logs. Any specific reason you added this?

**Review by fmartingr — APPROVED** [review]
LGTM. Just a question about storing log files as artifacts.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Monirzadeh) [reply .github/workflows/ci.yml]: i follow documentation no specific reasion. now i remove that.
is it what you want?

**Review by fmartingr — APPROVED** [review]

**Pull request merged**